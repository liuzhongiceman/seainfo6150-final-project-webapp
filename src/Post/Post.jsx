import React, { useState, useEffect } from "react";
import styles from "./Post.module.css";


class Post extends React.Component {
  
  fetchPost = () => {
    const post = this.props.posts.find((post) => post.id = this.props.postId);
    if (post) {
      console.log('post', post);
      return (
        <div>
          <img className={styles.postImage} src={post.url} alt="post"/>
          <div className={styles.ratingContainer}>
            <div>${post.price} / night</div>
            <div className={styles.starContainer}>
                <img className={styles.star} src="https://www.flaticon.com/svg/static/icons/svg/1828/1828884.svg" alt="star"/>
                {post.rating}
            </div>
          </div>
          <h2>{post.title}</h2>
          <p className={styles.postDescription}>{post.description}</p>
        </div>
      )
    }
  }
  
  getCategoryImage = (category) => {
    switch(category) {
        case 'treehouse':
            return 'https://news.airbnb.com/wp-content/uploads/sites/4/2018/08/1-mushroom-dome-min.jpg?fit=1024,683&resize=2048,1366';
            break;
        case 'tinyhouse':
            return 'https://a0.muscache.com/im/pictures/81a84488-3297-4a9f-b9d1-6dcb3ae9e726.jpg?im_w=1440';
            break;
        case 'farmstay' :
            return 'https://a0.muscache.com/im/pictures/467632ae-09d8-4334-8620-0f67cb4fd055.jpg?im_w=1440';
            break;
        case 'boat' :
            return 'https://a0.muscache.com/im/pictures/00d3733e-a272-48d5-8601-d53170bdce97.jpg?im_w=1200';
            break;
        case 'camper' :
            return 'https://a0.muscache.com/im/pictures/dcb763ed-e6ff-4680-880e-f21026a584dc.jpg?im_w=1440';
            break;
        case 'tent' :
            return 'https://a0.muscache.com/im/pictures/1e4a48e2-91cc-404f-a9d4-5d676b009c49.jpg?im_w=1200';
            break;
        case 'yurt' :
            return 'https://a0.muscache.com/im/pictures/4cb336e9-c8b2-4337-90ec-a75d6005e938.jpg?im_w=1440';
            break;
    }
}
  
  displayCategories = () => {
    const listItems = this.props.categories.map((category, index) => 
            <a className={styles.category} href={`/categories/${category}`} key={index}>
                <img className={styles.categoryImg} src={this.getCategoryImage(category)} />
                <div className={styles.categoryBottom}>
                    <span className={styles.categoryTitle}>{category}</span>
                </div>
            </a>
    );
    return (listItems)
}
  
  
  render() {
    return (
      <div>
        <div className={styles.row2}>
          <h2 className={styles.anywhere}>Live anywhere</h2>
          <div className={styles.row2Container}>
              {this.displayCategories()}
          </div>
        </div>
        <div className={styles.postMain}>
          {this.fetchPost()}
        </div>
      </div>
    )
  }
}

export default Post
