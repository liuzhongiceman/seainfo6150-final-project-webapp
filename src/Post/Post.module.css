.postMain {
  padding-top: 100px;
  max-width: 1200px;
  margin: 0 auto;
  padding-bottom: 100px;
}

.postImage {
  max-width: 800px;
  width: 100%;
}

.postDescription {
  width: 100%;
  line-height: 20px;
  color: #1A2045;
}

.row2 {
  padding-top: 50px;
  width: 100%;
  display: flex;
  flex-direction: column;
  max-width: 1800px;
  margin: 0 auto;
}

.anywhere { margin-left: 10px; }

.row2Container {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
}

.categoryImg {
  width: 100%;
  height: 168px;
  border-radius: 25px;
}

.category {
  margin: 10px;
  width: 20%;
}

.categoryBottom {
  margin-top: 10px;
  text-align: center;
  text-transform: capitalize;
}

.categoryTitle {
  color: #1A2045;
  font-weight: 600;
  text-transform: capitalize;
}

.star {
  width: 20px;
  height: 20px;
  margin-right: 10px;
  color: #DC143C;
}

.title {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
}

.starContainer {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
}

.ratingContainer {
  padding-top: 30px;
  max-width: 800px;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
}

@media screen and (min-width: 1200px) {
  
  .category {
    margin: 10px;
    width: 10%;
  }
  
  .postDescription {
    width: 60%;
  }
}