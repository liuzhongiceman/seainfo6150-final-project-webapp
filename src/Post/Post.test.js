import React from "react";
import { render } from "@testing-library/react";
import Post from "./Post.jsx";

it("renders the Post component correctly", () => {
  
  const categories = ["treehouse", "tinyhouse", "farmstay", "boat", "camper", "tent", "yurt"];
  const postId = "camper2";
  
  const posts = [ {
      description: "Surrounded by vineyards and walnut orchards in the rugged Santa Lucia mountains west of Paso Robles, you will feel as though you're hours away from humanity, yet you’re just 10 min from downtown Paso Robles and about 1/2 an hour from several beach towns. Also, the Airstream shares the property with a winery and there are four others minutes away.",
      id: "camper2",
      price: 101,
      rating: 4.97,
      subtitle: "4 guests · 2 bedrooms · 2 beds · 1 bath",
      title: "Wine Country Airstream Experience",
      url: "https://a0.muscache.com/im/pictures/f6336eb2-30ca-49d4-9ea4-fdf640409e32.jpg?im_w=1200"
    }
  ]
  
  
  const { container } = render(<Post posts={posts} postId={postId} categories={categories}/>);
  expect(container).toMatchSnapshot();
});
