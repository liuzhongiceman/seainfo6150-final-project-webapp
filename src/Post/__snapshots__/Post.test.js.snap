// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Post component correctly 1`] = `
<div>
  <div>
    <div
      class="row2"
    >
      <h2
        class="anywhere"
      >
        Live anywhere
      </h2>
      <div
        class="row2Container"
      >
        <a
          class="category"
          href="/categories/treehouse"
        >
          <img
            alt="category"
            class="categoryImg"
            src="https://news.airbnb.com/wp-content/uploads/sites/4/2018/08/1-mushroom-dome-min.jpg?fit=1024,683&resize=2048,1366"
          />
          <div
            class="categoryBottom"
          >
            <span
              class="categoryTitle"
            >
              treehouse
            </span>
          </div>
        </a>
        <a
          class="category"
          href="/categories/tinyhouse"
        >
          <img
            alt="category"
            class="categoryImg"
            src="https://a0.muscache.com/im/pictures/81a84488-3297-4a9f-b9d1-6dcb3ae9e726.jpg?im_w=1440"
          />
          <div
            class="categoryBottom"
          >
            <span
              class="categoryTitle"
            >
              tinyhouse
            </span>
          </div>
        </a>
        <a
          class="category"
          href="/categories/farmstay"
        >
          <img
            alt="category"
            class="categoryImg"
            src="https://a0.muscache.com/im/pictures/467632ae-09d8-4334-8620-0f67cb4fd055.jpg?im_w=1440"
          />
          <div
            class="categoryBottom"
          >
            <span
              class="categoryTitle"
            >
              farmstay
            </span>
          </div>
        </a>
        <a
          class="category"
          href="/categories/boat"
        >
          <img
            alt="category"
            class="categoryImg"
            src="https://a0.muscache.com/im/pictures/00d3733e-a272-48d5-8601-d53170bdce97.jpg?im_w=1200"
          />
          <div
            class="categoryBottom"
          >
            <span
              class="categoryTitle"
            >
              boat
            </span>
          </div>
        </a>
        <a
          class="category"
          href="/categories/camper"
        >
          <img
            alt="category"
            class="categoryImg"
            src="https://a0.muscache.com/im/pictures/dcb763ed-e6ff-4680-880e-f21026a584dc.jpg?im_w=1440"
          />
          <div
            class="categoryBottom"
          >
            <span
              class="categoryTitle"
            >
              camper
            </span>
          </div>
        </a>
        <a
          class="category"
          href="/categories/tent"
        >
          <img
            alt="category"
            class="categoryImg"
            src="https://a0.muscache.com/im/pictures/1e4a48e2-91cc-404f-a9d4-5d676b009c49.jpg?im_w=1200"
          />
          <div
            class="categoryBottom"
          >
            <span
              class="categoryTitle"
            >
              tent
            </span>
          </div>
        </a>
        <a
          class="category"
          href="/categories/yurt"
        >
          <img
            alt="category"
            class="categoryImg"
            src="https://a0.muscache.com/im/pictures/4cb336e9-c8b2-4337-90ec-a75d6005e938.jpg?im_w=1440"
          />
          <div
            class="categoryBottom"
          >
            <span
              class="categoryTitle"
            >
              yurt
            </span>
          </div>
        </a>
      </div>
    </div>
    <div
      class="postMain"
    >
      <div>
        <img
          alt="post"
          class="postImage"
          src="https://a0.muscache.com/im/pictures/f6336eb2-30ca-49d4-9ea4-fdf640409e32.jpg?im_w=1200"
        />
        <div
          class="ratingContainer"
        >
          <div>
            $
            101
             / night
          </div>
          <div
            class="starContainer"
          >
            <img
              alt="star"
              class="star"
              src="https://www.flaticon.com/svg/static/icons/svg/1828/1828884.svg"
            />
            4.97
          </div>
        </div>
        <h2>
          Wine Country Airstream Experience
        </h2>
        <p
          class="postDescription"
        >
          Surrounded by vineyards and walnut orchards in the rugged Santa Lucia mountains west of Paso Robles, you will feel as though you're hours away from humanity, yet youâ€™re just 10 min from downtown Paso Robles and about 1/2 an hour from several beach towns. Also, the Airstream shares the property with a winery and there are four others minutes away.
        </p>
      </div>
    </div>
  </div>
</div>
`;
